### PR TITLE
github: update CODEOWNERS

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #4399.

### What is changed and how does it work?

```commit-message
Update `CODEOWNERS` to add `/pkg/schedule/config/config.go`.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by HuSharp** [comment]
does `resource manager` config need to be protected?

**Review by rleungx — APPROVED** [review]

**Comment by JmPotato** [comment]
> resource manager

Good catch. Added.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
+# Require review from domain experts when the PR modified significant config files
+
+/conf/config.toml @tikv/pd-configuration-reviewer
 /server/config/config.go @tikv/pd-configuration-reviewer
+/pkg/schedule/config/config.go @tikv/pd-configuration-reviewer
 /pkg/schedule/schedulers/hot_region_config.go @tikv/pd-configuration-reviewer
-/conf/config.toml @tikv/pd-configuration-reviewer
+/pkg/mcs/resourcemanager/server/config.go @tikv/pd-configuration-reviewer
+/pkg/mcs/scheduling/server/config/config.go @tikv/pd-configuration-reviewer
+/pkg/mcs/tso/server/config.go @tikv/pd-configuration-reviewer
```
Comment (HuSharp) [inline .github/CODEOWNERS:9]: Preferably in lexicographical order :)

**Review by HuSharp — APPROVED** [review]
LGTM!

**Pull request merged**